core/cont/inc/TObjArray.h

```diff
@@ -134,6 +131,13 @@ class TObjArrayIter : public TIterator,
    TObjArrayIter() : fArray(0), fCurCursor(0), fCursor(0), fDirection(kIterForward) { }
 
 public:
+   using iterator_category = std::bidirectional_iterator_tag; // TODO: ideally it should be a randomaccess_iterator_tag
```

Comment:
@pcanal even the original code had this `TODO` comment. My argument here is: why going for `randomaccess_iterator_tag` when `bidirectional_iterator_tag` already does its job. Or should I change it?

Same remark applies for `TRefArrayIter`

Reply:
I guess, for `randomaccess_iterator_tag` one should support more functionality - like arithmetic with iterators `begin()+5`.
It is not provided by base `TCollection` class, therefore one need more changes to get it working with `TObjArray` properly. I guess, this PR should focus on fixing compiler warnings

Reply:
Then this PR should already be ready to go :)